src/back/spv/writer.rs

```diff
@@ -1017,14 +1017,29 @@ impl Writer {
                 let mut constituent_ids = Vec::with_capacity(components.len());
                 for component in components {
                     let expression = &ir_function.expressions[*component];
-                    let (component_id, _) = self.write_expression(
+                    let (component_id, component_local_ty) = self.write_expression(
                         ir_module,
                         &ir_function,
                         expression,
                         block,
                         function,
                     )?;
 
+                    let component_id = match expression {
```

Comment:
we should start abstracting this block away, given how often we have to repeat it

Reply:
Agreed, will add it to my todo